Project3 Front End

Adan-Asim-ASE opened the pull request:
Complete front end code of project3

Adan-Asim-ASE committed:
Complete front end

Adan-Asim-ASE committed:
Added line at end of file

kamranarshad1 commented on project3/.env:
```diff
@@ -0,0 +1 @@
+REACT_APP_BACKEND_API=  "http://localhost:8080/"
```
Please add `.env` file in `.gitignore`

kamranarshad1 commented on project3/src/App.js:
```diff
@@ -0,0 +1,32 @@
+
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<Main />} />
+        <Route path="/Login" element={<Login />} />
+        <Route path="/Home" element={<Home />} />
+        <Route path="/Home/Me/Published" element={<MyPublishedPosts />} />
+        <Route path="/Home/Me/Drafted" element={<MyDraftedPosts />} />
+        <Route path="/Home/PublicPosts" element={<PublicPosts/>} />
+        <Route path="/Signup" element={<Signup />} />
+        <Route path="/Post/Edit/:id" element={<Editpost />} />
+      </Routes>
```
Please create separate components for `Private` & `Public` routes

kamranarshad1 replied:
make all the routes kebab cased

kamranarshad1 commented on project3/src/App.js:
```diff
@@ -0,0 +1,32 @@
+        <Route path="/Home" element={<Home />} />
+        <Route path="/Home/Me/Published" element={<MyPublishedPosts />} />
+        <Route path="/Home/Me/Drafted" element={<MyDraftedPosts />} />
+        <Route path="/Home/PublicPosts" element={<PublicPosts/>} />
```
What's the difference b/w PublicPosts & Home?

Adan-Asim-ASE replied:
PublicPosts lists Public/Published Posts made by all users, while Home shows All posts(Both Published & Drafted) made by loggedIn user

kamranarshad1 commented on project3/src/App.js:
```diff
@@ -0,0 +1,32 @@
+        <Route path="/Home/Me/Drafted" element={<MyDraftedPosts />} />
+        <Route path="/Home/PublicPosts" element={<PublicPosts/>} />
+        <Route path="/Signup" element={<Signup />} />
+        <Route path="/Post/Edit/:id" element={<Editpost />} />
```
Should be `posts/:id/edit`

kamranarshad1 commented on project3/src/App.js:
```diff
@@ -0,0 +1,32 @@
+        <Route path="/Login" element={<Login />} />
+        <Route path="/Home" element={<Home />} />
+        <Route path="/Home/Me/Published" element={<MyPublishedPosts />} />
+        <Route path="/Home/Me/Drafted" element={<MyDraftedPosts />} />
```
Url should be `posts/me/drafts`

kamranarshad1 commented on project3/src/App.js:
```diff
@@ -0,0 +1,32 @@
+        <Route path="/" element={<Main />} />
+        <Route path="/Login" element={<Login />} />
+        <Route path="/Home" element={<Home />} />
+        <Route path="/Home/Me/Published" element={<MyPublishedPosts />} />
```
Url should be `posts/me/published`

kamranarshad1 commented on project3/src/Pages/Editpost.js:
```diff
@@ -0,0 +1,99 @@
+
+    let [title, setTitle] = useState(post.title);
+    let [content, setContent] = useState(post.content);
+    let [status, setStatus] = useState(post.published === true ? "Published" : "Drafted");
```
Create constants for statuses & avoid magic strings

Adan-Asim-ASE replied:
Actually, user can edit post and can change status from drafted to published and vice versa, therefore kept it let

Adan-Asim-ASE replied:
Actually, user can edit post and can change status from drafted to published and vice versa, therefore kept it let

Adan-Asim-ASE committed:
Added .env to gitIgnore

Adan-Asim-ASE committed:
App.js: Created separate components for Public and Protected Routes, changed URLs and names

Adan-Asim-ASE committed:
Few changes

Adan-Asim-ASE committed:
Changed .env variable

Adan-Asim-ASE committed:
.env: putted in gitignore

kamranarshad1 commented on project3/src/App.css:
```diff
@@ -0,0 +1,38 @@
+.App {
+  text-align: center;
+}
+
+.App-logo {
+  height: 40vmin;
+  pointer-events: none;
+}
+
+@media (prefers-reduced-motion: no-preference) {
+  .App-logo {
+    animation: App-logo-spin infinite 20s linear;
+  }
+}
+
+.App-header {
+  background-color: #282c34;
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
+  color: white;
+}
+
+.App-link {
+  color: #61dafb;
+}
+
```
Please remove unused css

kamranarshad1 commented on project3/src/Pages/Home.js:
```diff
@@ -0,0 +1,87 @@
+
+  useEffect(() => {
+
+    fetch(process.env.REACT_APP_BACKEND_API + "posts/me", config)
```
Please use one library for api calls, either fetch or axios

kamranarshad1 commented on project3/src/Pages/MyPublishedPosts.js:
```diff
@@ -0,0 +1,84 @@
+      authorization: 'Bearer ' + userToken,
+    }
+  };
+
```
Create a separate function for api requests, just pass it endpoint & data and api call should be managed by that function

kamranarshad1 commented on project3/src/Pages/PublicPosts.js:
```diff
@@ -0,0 +1,62 @@
+
+
+  useEffect(() => {
+
```
Please remove all extra lines in all useEffects

kamranarshad1 commented on project3/src/Pages/PublicRoute.js:
```diff
@@ -0,0 +1,5 @@
+const PublicRoute = ({ children }) => {
+  return children;
```
No logic required for Public routes? Then whats the purpose of this component?

kamranarshad1 replied:
This component should use `Route` inside

kamranarshad1 commented on project3/src/Pages/ProtectedRoute.js:
```diff
@@ -0,0 +1,14 @@
+import { Navigate } from 'react-router-dom';
+import getCookie from '../utils/HelperFunctions';
+
+const ProtectedRoute = ({ children }) => {
```
This component should use `Route` inside

kamranarshad1 commented on project3/src/Pages/Editpost.js:
```diff
@@ -0,0 +1,99 @@
+    setStatus(e.target.value)
+  }
+
+  async function submit(event) {
```
Please prefer using arrow functions in all places

Adan-Asim-ASE replied:
Acknowledged

Adan-Asim-ASE committed:
Removed unnecessary CSS
Replaced all fetch calls with axios
Separated API calls from pages
Solved naming and indentation issues
Resolved routing issues

Adan-Asim-ASE committed:
- Splitted the pages into smaller components.
- Configured ES lint and resolved issues highlighted by it.

Adan-Asim-ASE committed:
index.js, Helper Functions.js, reportWebVitals.js: Resolved minor issues highlighed by ESlint
